inc/neopixel.h

```diff
@@ -12,31 +12,49 @@ static void neopixel_send_buffer_core(volatile uint32_t *clraddr, uint32_t pinMa
                                       const uint8_t *ptr, int numBytes) {
     asm volatile("        push    {r4, r5, r6, lr};"
                  "        add     r3, r2, r3;"
-                 "        cpsid   i;"
                  "loopLoad:"
                  "        ldrb r5, [r2, #0];" // r5 := *ptr
                  "        add  r2, #1;"       // ptr++
                  "        movs    r4, #128;"  // r4-mask, 0x80
                  "loopBit:"
                  "        str r1, [r0, #4];"                    // set
+                 #ifdef SAMD21
                  "        movs r6, #3; d2: sub r6, #1; bne d2;" // delay 3
+                 #endif
+                 #ifdef SAMD51
+                 "        movs r6, #3; d2: subs r6, #1; bne d2;" // delay 3
```

Comment:
I thought there is only `subs` in Thumb instruction set (ie the instruction always sets the flags) - doesn't SAMD21 accept this?

Reply:
I didn't write that code, but testing:
```
$ cat sub.c
void f(void) {
    asm volatile("sub r6, #1");
}
$ cat subs.c
void f(void) {
    asm volatile("subs r6, #1");
}
$ arm-none-eabi-gcc -mthumb -mcpu=cortex-m0plus -c sub.c
$ arm-none-eabi-gcc -mthumb -mcpu=cortex-m4 -c sub.c
$ arm-none-eabi-gcc -mthumb -mcpu=cortex-m0plus -c subs.c
/tmp/ccSbIfJ3.s: Assembler messages:
/tmp/ccSbIfJ3.s:27: Error: instruction not supported in Thumb16 mode -- `subs r6,#1'
$ arm-none-eabi-gcc -mthumb -mcpu=cortex-m4 -c subs.c
```
This seems to be a toolchain bug:
https://forum.pjrc.com/threads/27624-Coming-Soon-Teensy-LC-(low-cost-Teensy)/page13
https://sourceware.org/bugzilla/show_bug.cgi?id=15348